### ISSUE TITLE
GradMonitor doesn't work with deepspeed

### ISSUE THREAD

**Issue by ananyahjha93** [mosaicml/composer#1501]
** Environment **

- OS: Ubuntu 20.04
- Hardware (GPU, or instance type): 8xA100
- cuda: 11.3
- cudnn: 8
- pytorch: 1.12.1
- composer: dev branch installed from source
- deepspeed: 0.7.2
- transformers: 4.21.2

** To reproduce

Use gradmonitor with and without deepspeed.

When I train the smaller 560M bloom model without deepspeed this functions correctly. With deepspeed, for the larger 1.7B model gradmonitor outputs all 0s and doesn't show any layer-wise gradnorms.

Deepspeed works with fp16, without deepspeed (when gradmonitor works), I train with bf16.

![image](https://user-images.githubusercontent.com/7491256/188511352-7e0ad1c2-55c7-4f1a-949b-59a3bb07595b.png)
![image](https://user-images.githubusercontent.com/7491256/188511359-ed3e5049-1f59-4b79-8bc0-db0e58d0d85f.png)

Now to check if this is a model size issue or a logger issue, I run the smaller 560M model with and without deepspeed and use simple print statements to debug.

Debugging:
<img width="682" alt="Screen Shot 2022-09-05 at 1 54 17 PM" src="https://user-images.githubusercontent.com/7491256/188511666-25f7d4f8-5d28-4805-a76c-6336673e0cf3.png">

Without deepspeed, grad norms are printed for each layer.
<img width="463" alt="Screen Shot 2022-09-05 at 1 54 23 PM" src="https://user-images.githubusercontent.com/7491256/188511670-151956a6-15d7-456b-8c92-b64536d60373.png">

With deepspeed, no grad norms are printed.
<img width="277" alt="Screen Shot 2022-09-05 at 1 58 58 PM" src="https://user-images.githubusercontent.com/7491256/188511908-4007a26a-8f53-464b-8058-49c8f05015ac.png">

**Comment by bandish-shah** [comment]
Hi @ananyahjha93, thanks for reporting this!  We'll look into it and get back to you soon.

**Comment by bcui19** [comment]
Hi @ananyahjha93, apologies for the delay I was digging into this issue, I believe it's an issue with deepspeed see:

https://github.com/microsoft/DeepSpeed/issues/2329

tl;dr deepspeed with zero optimizer stage 1 and 'n' gradient accumulation steps clears gradients in the computation graph at the 'nth' backwards call. As a result, the true minibatch gradient is inaccessible when using deepspeed zero optimizer stage 1. 

As a result this right now is a do not support.

**Comment by hanlint** [comment]
Closing this as a `wontfix` due to lack of support within DeepSpeed. We'll re-open if deepspeed allows for grad introspection!
